File: src/main/kotlin/fiftyPopularCodintInterviewProblems/MaximumSubarray.java
package fiftyPopularCodintInterviewProblems;

public class MaximumSubarray {
    public int maximumSubarray(int[] arr){
        int oldmax = arr[0];
        int max = 0;
        for (int i = 0; i < arr.length; i++) {
            max = 0;
            for (int j = i; j < arr.length; j++) {
                max += arr[j];
                if (max > oldmax)
                    oldmax = max;
            }
        }
        System.out.println(max);
        return oldmax;
    }
}

File: src/main/kotlin/leetCode/medium/MaximumSubarray.kt
package leetCode.medium

class MaximumSubarray {
    fun maxSubArray(nums: IntArray): Int {
        var gSum = Int.MIN_VALUE
        var lSum = 0
        for (i in 0..nums.size-1) {
            lSum += nums[i]
            gSum = Math.max(gSum, lSum)
            if (lSum < 0)
                lSum = 0
        }
        return gSum
    }
}